File: Second year/MPP/Java_MPP_Bun/src/main/java/com/example/java_mpp_bun/repository/Repository.java
package com.example.java_mpp_bun.repository;



import com.example.java_mpp_bun.domain.Entity;

import java.time.LocalDateTime;
import java.util.List;

public interface Repository<ID, E extends Entity<ID>> {

    E findOne(ID id);
    List<E> findByDestination(String destinatie, LocalDateTime data);
    E findByName(String userName);
    Iterable<E> findAll();
    void save(E entity);
    void update( E entity);

}

File: Second year/MPP/C#/Incercare/repository/Repository.cs
using System;
using System.Collections.Generic;

namespace Incercare.repository
{
    public interface Repository<TId, TEntity>
    {
        List<TEntity> findAll();
        TEntity findOne(int id);
        TEntity findByName(string nume);
        TEntity findByDestination(string destinatie, DateTime data);
        TEntity save(TEntity entity);
        TEntity update(TEntity entity);
    }
}